libs/pages/services/src/lib/feature/page-settings-preview-environments-feature/page-settings-preview-environments-feature.tsx

```diff
@@ -42,34 +44,33 @@ export function PageSettingsPreviewEnvironmentsFeature() {
 
   const onSubmit = methods.handleSubmit(async (data) => {
     if (data) {
+      setLoading(true)
       // update auto preview for environment
       const cloneEnvironmentDeploymentRules = Object.assign({}, environmentDeploymentRules as EnvironmentDeploymentRule)
       cloneEnvironmentDeploymentRules.auto_preview = data['auto_preview']
 
-      if (environmentDeploymentRules?.auto_preview !== data['auto_preview']) {
-        await dispatch(
-          editEnvironmentDeploymentRules({
-            environmentId,
-            deploymentRuleId: environmentDeploymentRules?.id || '',
-            data: cloneEnvironmentDeploymentRules,
-          })
-        )
-      }
+      await dispatch(
+        editEnvironmentDeploymentRules({
+          environmentId,
+          deploymentRuleId: environmentDeploymentRules?.id || '',
+          data: cloneEnvironmentDeploymentRules,
+        })
+      )
 
       applications?.forEach(async (application: Application) => {
         if (application.id === Object.keys(data).find((key) => key === application.id)) {
           const cloneApplication = Object.assign({}, application as Application)
           cloneApplication.auto_preview = data[application.id]
-          if (application.auto_preview !== data[application.id]) {
-            await dispatch(
-              editApplication({
-                applicationId: application.id,
-                data: cloneApplication,
-              })
-            )
-          }
+          await dispatch(
```

Comment:
here I don't see a unwrap / then / catch. The edit is always working ? I guess yes since values comes from select and there are not too much strange validation backend side. 
If the edit has 90% chance to work all the time, we can then skip the unwrap / then / catch 
Is it something you agree with?